apex-squid/src/test/java/org/fundacionjala/enforce/sonarqube/apex/parser/grammar/ApexGrammarDmlUpsertTest.java

```diff
@@ -23,34 +23,30 @@
  */
 package org.fundacionjala.enforce.sonarqube.apex.parser.grammar;
 
+import org.fundacionjala.enforce.sonarqube.apex.parser.ApexRuleTest;
 import org.junit.Before;
 import org.junit.Test;
-
-import org.fundacionjala.enforce.sonarqube.apex.parser.ApexRuleTest;
-
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.DML_UPSERT;
 import static org.sonar.sslr.tests.Assertions.assertThat;
 
-import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.STATEMENT_BLOCK;
-
-public class ApexGrammarStatementBlockTest extends ApexRuleTest {
+public class ApexGrammarDmlUpsertTest extends ApexRuleTest {
 
     @Before
     public void init() {
-        setRootRule(STATEMENT_BLOCK);
+        setRootRule(DML_UPSERT);
     }
 
     @Test
-    public void RulesStatementBlock() {
+    public void rulesDmlUpsert() {
         assertThat(parser)
-                .matches("{}")
-                .matches("{int myVariable;}")
-                .matches("{int myVariable=23;if(NAME)12;}");
+                .matches("upsert primaryExpression")
```

Comment:
this would be a valid scenario too

upsert primaryExpression

upsert primaryExpression;